### ISSUE TITLE
'vet' linter not detecting lock copy

### ISSUE THREAD

**Issue by leonklingele** [alecthomas/gometalinter#464]
## Steps to reproduce

```bash
$ go get -u github.com/leonklingele/github-release-checker
$ cd $GOPATH/src/github.com/leonklingele/github-release-checker
$ git checkout fasthttp
```

Apply the following patch which causes a lock copy:

```patch
diff --git a/checker/github/tag/list.go b/checker/github/tag/list.go
index 1263edb..72388cb 100644
--- a/checker/github/tag/list.go
+++ b/checker/github/tag/list.go
@@ -19,7 +19,7 @@ const (
 	maxWorkers = 100
 )
 
-func newListWorker(repoChan repository.Chan, tagChan chanW, c *fasthttp.Client) {
+func newListWorker(repoChan repository.Chan, tagChan chanW, c fasthttp.Client) {
 	ts := time.Now().Unix()
 	fp := gofeed.NewParser()
 	for repo := range repoChan {
@@ -59,7 +59,7 @@ func List(repoChan repository.Chan) Chan {
 			logging.Debug("done listing tags")
 		}()
 
-		c := &fasthttp.Client{}
+		c := fasthttp.Client{}
 		for i := 0; i < maxWorkers; i++ {
 			wg.Add(1)
 			go func() {

```

## Expected result

`go vet` correctly detects the issue:

```bash
$ go vet ./...
# ./github-release-checker/checker/github/tag
checker/github/tag/list.go:22: newListWorker passes lock by value: github.com/valyala/fasthttp.Client contains github.com/valyala/fasthttp.noCopy
checker/github/tag/list.go:67: call of newListWorker copies lock value: github.com/valyala/fasthttp.Client contains github.com/valyala/fasthttp.noCopy
```

## Actual result

.. `gometalinter` doesn't:

```bash
$ gometalinter --vendor --disable-all --enable vet --debug ./...
DEBUG: [Apr 25 00:58:05.519] PATH=mypath
DEBUG: [Apr 25 00:58:05.519] GOPATH=/path/to/go
DEBUG: [Apr 25 00:58:05.519] GOBIN=
DEBUG: [Apr 25 00:58:05.519] linting path .
DEBUG: [Apr 25 00:58:05.519] linting path ./checker
DEBUG: [Apr 25 00:58:05.519] linting path ./checker/github
DEBUG: [Apr 25 00:58:05.519] linting path ./checker/github/repository
DEBUG: [Apr 25 00:58:05.519] linting path ./checker/github/tag
DEBUG: [Apr 25 00:58:05.519] linting path ./checker/handlers
DEBUG: [Apr 25 00:58:05.519] linting path ./checker/handlers/log
DEBUG: [Apr 25 00:58:05.519] linting path ./checker/handlers/mail
DEBUG: [Apr 25 00:58:05.519] linting path ./config
DEBUG: [Apr 25 00:58:05.519] linting path ./logging
DEBUG: [Apr 25 00:58:05.519] linting path ./pathutil
DEBUG: [Apr 25 00:58:05.519] linting path ./stringutil
DEBUG: [Apr 25 00:58:05.524] [vet.1]: executing /path/to/go/bin/govet --no-recurse . ./checker ./checker/github ./checker/github/repository ./checker/github/tag ./checker/handlers ./checker/handlers/log ./checker/handlers/mail ./config ./logging ./pathutil ./stringutil
DEBUG: [Apr 25 00:58:05.608] [vet.1]: vet hits 0: ^(?:vet:.*?\.go:\s+(?P<path>.*?\.go):(?P<line>\d+):(?P<col>\d+):\s*(?P<message>.*))|(?:(?P<path>.*?\.go):(?P<line>\d+):\s*(?P<message>.*))$
DEBUG: [Apr 25 00:58:05.608] [vet.1]: vet linter took 83.986607ms
DEBUG: [Apr 25 00:58:05.608] total elapsed time 89.351436ms
```

It works fine if the project is cloned outside of `$GOPATH`:

```bash
$ cd $(mktemp -d)
$ git clone https://github.com/leonklingele/github-release-checker -b fasthttp
$ cd github-release-checker
$ # Apply the patch from above
$ gometalinter --vendor --disable-all --enable vet ./...
checker/github/tag/list.go:22::error: newListWorker passes lock by value: github.com/valyala/fasthttp.Client contains github.com/valyala/fasthttp.noCopy (vet)
checker/github/tag/list.go:67::error: function call copies lock value: github.com/valyala/fasthttp.Client contains github.com/valyala/fasthttp.noCopy (vet)
```

Hmm.. Maybe https://github.com/dnephin/govet needs an update?

**Comment by leonklingele** [comment]
@dnephin thanks, indeed fixed by #459. What's blocking the merge?

**Comment by dnephin** [comment]
Thanks for confirming the fix. I'll close this issue and go ahead and merge #459 .
